toolkit/scss/_buttons.scss

```diff
@@ -27,6 +27,10 @@ input.button-save-with-advice {
   }
 }
 
+input.button-with-temporary-message {
```

Comment:
The name seems odd here - is it not more like "button-inside-temporary-message"?  And in that case, rather than applying the class to the button it would make more sense to me to add the additional margin as part of the temporary message styling, rather than as part of the button styling, the same as you've added the special link styling for links in `_temporary-message.scss`.

So have no new class name at all (i.e. no "button-with-temporary-message" class), but instead have special styling for buttons inside a temporary-message.